cli/src/commands/help.rs

```diff
@@ -28,22 +32,98 @@ pub(crate) struct HelpArgs {
 
 #[instrument(skip_all)]
 pub(crate) fn cmd_help(
-    _ui: &mut Ui,
+    ui: &mut Ui,
     command: &CommandHelper,
     args: &HelpArgs,
 ) -> Result<(), CommandError> {
+    // TODO: Add all documentation to categories
+    //
+    // Maybe adding some code to build.rs to find all the docs files and build the
+    // BTreeMap at compile time.
+    //
+    // It would be cool to follow the docs hierarchy somehow.
+    //
+    // One of the problems would be `config.md`, as it has the same name as a
+    // subcommand.
+    let categories = BTreeMap::from([
+        (
+            "revsets",
+            Category {
+                description: "A functional language for selecting a set of revision",
+                content: include_str!("../../../docs/revsets.md"),
+            },
+        ),
+        (
+            "tutorial",
+            Category {
+                description: "Show a tutorial to get started with jj",
+                content: include_str!("../../../docs/tutorial.md"),
+            },
+        ),
+    ]);
+
+    if let [name] = &*args.command {
+        if let Some(category) = categories.get(name.as_str()) {
+            ui.request_pager();
+            write!(ui.stdout(), "{}", category.content)?;
+
+            return Ok(());
+        }
+    }
+
     let mut args_to_show_help = vec![command.app().get_name()];
     args_to_show_help.extend(args.command.iter().map(|s| s.as_str()));
     args_to_show_help.push("--help");
 
+    let help_template_before_categories = "\
+    {about-with-newline}\n{usage-heading} {usage}\n\n\x1b[1m\x1b[4mCommands:\x1b[0m\n{subcommands}
+    ";
+
+    let subcommand_max_len = command
+        .app()
+        .get_subcommands()
+        .map(|cmd| cmd.get_name().len())
+        .max()
+        .unwrap();
+
+    let help_template_categories = format!(
+        "\x1b[1m\x1b[4mHelp Categories:\x1b[0m\n{}",
```

Comment:
This is pretty hard to read. It's making the text bold, right? Can we use the formatter (`ui.stdout_formatter()`) instead? Then it will also respect `NO_COLOR` and `ui.color`.

Reply:
Wouldn't be better to use `ui.new_formatter()`?

Reply:
Oh, you just want a string/bytes here? Yes, `ui.new_formatter()` is better then.

Reply:
It's probably easier to use crossterm API (or clap's styling functions if available.) clap will strip ANSI escapes when color is disabled.
https://docs.rs/crossterm/latest/crossterm/style/trait.Stylize.html